pkg/policies/controlplane/policy-service.go

```diff
@@ -308,30 +308,22 @@ func (s *PolicyService) GetDynamicConfig(ctx context.Context, req *policylangv1.
 
 	resp, err := s.etcdClient.KV.Get(ctx, path.Join(paths.PoliciesAPIDynamicConfigPath, req.PolicyName))
 	if err != nil {
-		return nil, fmt.Errorf("failed to write dynamic config '%s' to etcd: '%s'", req.PolicyName, err)
+		return nil, fmt.Errorf("failed to get dynamic config '%s' to etcd: '%s'", req.PolicyName, err)
 	}
 
-	configs := []*policylangv1.GetDynamicConfigResponse{}
-	for _, v := range resp.Kvs {
-
-		dynamicConfigJSON := make(map[string]interface{})
-		err = json.Unmarshal(v.Value, &dynamicConfigJSON)
-		if err != nil {
-			return nil, fmt.Errorf("failed to parse DynamicConfig JSON: %w", err)
-		}
-
-		dynamicConfigStruct, err := structpb.NewStruct(dynamicConfigJSON)
-		if err != nil {
-			return nil, fmt.Errorf("failed to parse DynamicConfig Struct: %w", err)
-		}
+	dynamicConfigJSON := make(map[string]interface{})
+	err = json.Unmarshal(resp.Kvs[0].Value, &dynamicConfigJSON)
```

Comment:
We should check if the len == 1 here else we might get out of index error here

Reply:
Added a check